### PR TITLE
Added SplitTensorsTransform & Auto-device-picker

### PR DESCRIPTION
**Pull request by irajagop**
- Refactor Transforms to take any kwargs
- Added SplitTensorsTransform to split the onnx file into multiple tensors file with following defaults:
  - file chunk size = 10 GB
  - threshold for external data = 1024 bytes (1 KB)
- Simplify onnx tests
- Updated `cloud_infer.py` to support auto-device-picker and other fixes

### CODE REVIEW

File: QEfficient/generation/cloud_infer.py [hunk 1]
```diff
@@ -58,9 +58,13 @@ def __init__(
...
-        devices = qaicrt.QIDList(device_ids)
-        self.context = qaicrt.Context(devices)
-        self.queue = qaicrt.Queue(self.context, device_ids[0])  # Async API
+        if device_ids is not None:
+            devices = qaicrt.QIDList(device_ids)
+            self.context = qaicrt.Context(devices)
+            self.queue = qaicrt.Queue(self.context, device_ids[0])
+        else:
+            self.context = qaicrt.Context()
+            self.queue = qaicrt.Queue(self.context, 0)  # Async API
```
Comment (ochougul) [inline QEfficient/generation/cloud_infer.py:67]: can you please explain, does this choose device id = 1, 2, 3, 4 when 0 is busy?

Comment (ochougul) [reply QEfficient/generation/cloud_infer.py:67]: Also, should we make `device_id` parameter optional in CLI and python APIs with this change?

Comment (irajagop) [reply QEfficient/generation/cloud_infer.py:67]: Done

File: QEfficient/cloud/execute.py [hunk 2]
```diff
@@ -16,7 +16,7 @@
 def main(
     model_name: str,
     qpc_path: str,
-    device_group: List[int],
+    device_group: Optional[List[int]] = None,
```
Comment (ochougul) [inline QEfficient/cloud/execute.py:19]: Please update docstring

File: QEfficient/base/onnx_transforms.py [hunk 3]
```diff
@@ -53,9 +53,32 @@ def apply(cls, model: ModelProto, onnx_base_dir: Optional[str] = None) -> Tuple[
...
-    pass
+    @classmethod
+    def apply(
+        cls,
+        model: ModelProto,
+        *,
+        model_name: str,
+        onnx_base_dir: Optional[str] = None,
+        file_chunk_size: int = 10 * 2**30,  # 10 GiB
+        size_threshold: int = 1024,
```
Comment (ochougul) [inline QEfficient/base/onnx_transforms.py:69]: What is the purpose of `size_threshold`?

File: QEfficient/cloud/execute.py (hunk 2 above)
Comment (irajagop) [reply QEfficient/cloud/execute.py:19]: Done

File: QEfficient/base/onnx_transforms.py (hunk 3 above)
Comment (irajagop) [reply QEfficient/base/onnx_transforms.py:72]: Only tensors greater than this threshold (in bytes) will be saved externally.

Added docstring to clarify the same.

File: QEfficient/compile/compile_helper.py [hunk 4]
```diff
@@ -40,7 +40,7 @@ def compile_kv_model_on_cloud_ai_100(
     custom_io_path: str,
     aic_enable_depth_first: bool,
     mos: int = -1,
-    device_group: List[int] = [0],
+    device_group: Optional[List[int]] = None,
```
Comment (ochougul) [inline QEfficient/compile/compile_helper.py:43]: please update docstring docstring

Comment (irajagop) [reply QEfficient/compile/compile_helper.py:43]: Updated for compile function.

**Pull request merged**